base/compiler/abstractinterpretation.jl

```diff
@@ -1141,7 +1141,7 @@ function const_prop_methodinstance_heuristic(interp::AbstractInterpreter,
     # was able to cut it down to something simple (inlineable in particular).
     # If so, there's a good chance we might be able to const prop all the way
     # through and learn something new.
-    if isdefined(method, :source) && is_inlineable(method.source)
+    if isdefined(method, :source) && method.source !== nothing && is_inlineable(method.source)
```

Comment:
seems like something is wrong if the `source` is defined but somehow also corrupted?

Reply:
Oh oops, I didn't intend to include this in this PR.